### PR TITLE
Add standalone executables

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

This keeps the `ign.cc` and `ign.hh` the same, and just provides a C++ entrypoint. This will help with the Windows port https://github.com/ignitionrobotics/ign-launch/pull/120

## Checklist
- [x] Signed all commits for DCO
- [x] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by scpeters** [comment]
I saw some code in the deleted `.rb.in` file about variable binding for `erb` with `:=` syntax. I don't think we have a test for that and haven't tested but am guessing it isn't yet supported by this pull request

**Comment by ahcorde** [comment]
@scpeters it's supported here

https://github.com/ignitionrobotics/ign-launch/blob/91427dc559904bb6b6c48f24903ddfbf35c375b7/src/cmd/launch_main.cc#L84-L102

**Comment by scpeters** [comment]
> @scpeters it's supported here
> 
> https://github.com/ignitionrobotics/ign-launch/blob/91427dc559904bb6b6c48f24903ddfbf35c375b7/src/cmd/launch_main.cc#L84-L102

great work!

File: src/cmd/cmdlaunch.rb.in [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+require 'open3'
+
+# Constants.
+LIBRARY_VERSION = '@PROJECT_VERSION_FULL@'
+COMMANDS = {
+  "launch" => "@launch_exe_location@",
+}
+
+#
+# Class for the Ignition transport command line tools.
```
Comment (mjcarroll) [inline src/cmd/cmdlaunch.rb.in:26]:
```suggestion
# Class for the Ignition launch command line tools.
```

File: src/cmd/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    PROPERTIES
+    ENVIRONMENT
+      "IGN_CONFIG_PATH=${CMAKE_BINARY_DIR}/test/conf/$<CONFIG>"
+  )
+endif()
+
+#===============================================================================
+# Generate the ruby script for internal testing.
+# Note that the major version of the library is included in the name.
+# Ex: cmdtransport0.rb
```
Comment (mjcarroll) [inline src/cmd/CMakeLists.txt:56]:
```suggestion
# Ex: cmdlaunch1.rb
```

File: src/cmd/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+file(GENERATE
+  OUTPUT "${cmd_script_generated_test}"
+  INPUT  "${cmd_script_configured_test}"
+)
+
+#===============================================================================
+# Used for the installed version.
+# Generate the ruby script that gets installed.
+# Note that the major version of the library is included in the name.
+# Ex: cmdtransport0.rb
```
Comment (mjcarroll) [inline src/cmd/CMakeLists.txt:79]:
```suggestion
# Ex: cmdlaunch1.rb
```

File: src/cmd/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+file(GENERATE
+  OUTPUT "${cmd_script_generated}"
+  INPUT  "${cmd_script_configured}")
+
+set(ign_library_path "${CMAKE_BINARY_DIR}/test/lib/ruby/ignition/cmd${IGN_DESIGNATION}${PROJECT_VERSION_MAJOR}")
+
+# Generate a configuration file for internal testing.
+# Note that the major version of the library is included in the name.
+# Ex: transport0.yaml
```
Comment (mjcarroll) [inline src/cmd/CMakeLists.txt:102]:
```suggestion
# Ex: launch1.yaml
```

File: src/cmd/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+file(GENERATE
+  OUTPUT "${CMAKE_BINARY_DIR}/test/conf/${IGN_DESIGNATION}${PROJECT_VERSION_MAJOR}.yaml"
+  INPUT "${CMAKE_CURRENT_BINARY_DIR}/${IGN_DESIGNATION}${PROJECT_VERSION_MAJOR}.yaml.configured")
+
+# Used for the installed version.
+set(ign_library_path "${CMAKE_INSTALL_PREFIX}/lib/ruby/ignition/cmd${IGN_DESIGNATION}${PROJECT_VERSION_MAJOR}")
+
+# Generate the configuration file that is installed.
+# Note that the major version of the library is included in the name.
+# Ex: transport0.yaml
```
Comment (mjcarroll) [inline src/cmd/CMakeLists.txt:116]:
```suggestion
# Ex: launch1.yaml
```

File: src/cmd/launch_main.cc [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    for (auto & arg : _opt.more_comms)
+    {
+      auto tokens = ignition::common::split(arg, ":=");
+      if (tokens.size() == 2)
+      {
+        cmd += " " + tokens[0] + "=" + tokens[1];
+      }
+      else
+      {
+        ignerr << "This argument [" << arg << "] is wrong for erb"
```
Comment (mjcarroll) [inline src/cmd/launch_main.cc:94]: Maybe a more descriptive error message here?

File: src/cmd/launch_main.cc [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+               << std::endl;
+      }
+    }
+    cmd += " " + _opt.launchfile;
+    std::string config = erbExec(cmd);
+    run(config.c_str());
+    return;
+  }
+
+  if (_opt.command == LaunchCommand::kNone)
```
Comment (mjcarroll) [inline src/cmd/launch_main.cc:104]: Add a test to cover this case.

**Review by mjcarroll — COMMENTED** [review]
Looks really good.  A few suggestions of `transport` to `launch` and maybe a place where we can increase coverage more.

**Comment by ahcorde** [comment]
@osrf-jenkins retest this please

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**